### DISCUSSION TITLE
YoLo v8 core dump at inference stage

### DISCUSSION THREAD

**Discussion by aldeponk** [ultralytics/ultralytics#19500] [Q&A]
Hi all,
I have a systematic problem of core dump at inference stage.
The context is I have a trained model that is perfectly working over Python3.8/torch2.1.1. 
I have to migrate to python 3.1x, preferably 3.12 at least, 3.13 if possible. To do so I synchronize torch version respectively Python3.12/torch2.5.0, Python3.13/torch2.6.0.
Unfortunately there is not much to give as error feedback unless that it goes into a core dump.
Is there any compatibility matrix I have missed ?
Thank you very much for any kind of help.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @aldeponk, thank you for reaching out and for using Ultralytics 🚀! We’re here to help. 

If this is a 🐛 Bug Report, we kindly ask you to provide additional details in the form of a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to assist with debugging. This helps us understand the issue and address it more effectively.

Regarding your migration to Python 3.12 or 3.13 with PyTorch 2.5.0/2.6.0, we recommend ensuring you’re using the latest Ultralytics package and verifying compatibility in the installation process. To upgrade to the latest version, ensure a clean and updated environment, and run:

```bash
pip install -U ultralytics
```

For reference, YOLO models are tested to work best within environments such as:
- **Python** ≥3.8
- **PyTorch** ≥1.8 (please check if any changes in PyTorch 2.5.0/2.6.0 affect compatibility)
- All other [dependencies](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)

If you haven’t already, you can test on our pre-configured environments:
- Notebooks with free GPU: 
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- [Docker Image](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/): <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Additionally, visit our [Docs](https://docs.ultralytics.com/) for helpful usage examples (e.g., [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/)) and [training tips](https://docs.ultralytics.com/guides/model-training-tips/) that may assist you with transition issues or custom configurations.

📢 For real-time answers, join our [Discord](https://discord.com/invite/ultralytics) community. For long-form discussion or troubleshooting, our [Discourse](https://community.ultralytics.com/) and [Subreddit](https://reddit.com/r/Ultralytics) are great options to connect with experts and other members.

This is an automated response to ensure you get a quick start! An Ultralytics engineer will follow up with you soon to assist further. 😊

**Comment by aldeponk** [comment]
Digging the documentation I just got aware that I can update ultralytics to latest version without any effect on a model trained with a previous version. Great ! 
Updating to latest + Python3.12/torch2.6 is working fine.
I close the thread.

**Reply by glenn-jocher** [reply]
@aldeponk thank you for the update! We're glad to hear updating to the latest Ultralytics version resolved your compatibility issues. For others facing similar core dump issues during inference, we recommend ensuring PyTorch/Torchvision versions align with our compatibility table in the checks utility and verifying CUDA compatibility. Always use the latest Ultralytics release for best support with newer Python/PyTorch versions. 

If any further issues arise, feel free to open a new GitHub issue with your environment details from `ultralytics.checks()`.
